### PR TITLE
fix: Several localisation issues.

### LINKED ISSUE #158: Localisation is not present on form validation.
Should probs be fixed using: https://github.com/jquense/yup?tab=readme-ov-file#localization-and-i18n

image

### LINKED ISSUE #165: `TransactionDetailModal` has placeholders/missing data
"Placeholders":

image
image

I assume the same also applies to `transactions.otherBoughtAt`.

### LINKED ISSUE #173: Form validation for title on POS creation also applied to owner input
image

### LINKED ISSUE #175: Date in mutation table not localised
Similar to GH-173 where localisation is not done properly.

In this case the "full" date is not localised based on the selected language:
https://github.com/GEWIS/sudosos-frontend-vue3/blob/cdc992f776cfd434833b2829d2feb1f90b4f2f5d/src/components/Mutations/MutationsTableComponent.vue#L17

The "short" date "is" (hardcoded):
https://github.com/GEWIS/sudosos-frontend-vue3/blob/cdc992f776cfd434833b2829d2feb1f90b4f2f5d/src/components/Mutations/MutationsTableComponent.vue#L18-L21

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
Fixes #175 
Fixes #173
Fixes #165 
Fixes #158

### CODE REVIEW

**Comment by SuperVK** [comment]
wait is this PR branched from #155 ?

**Comment by CodeNamedRobin** [comment]
@SuperVK could be? Wouldn't really matter I guess but I might have made a mistake when branching.

File: src/components/Mutations/MutationsTableComponent.vue [hunk 1]
```diff
@@ -14,9 +14,11 @@
           <Skeleton class="w-6 my-1 h-1rem surface-300"/>
         </template>
         <template #body="mutation" v-else>
-          <span class="hidden sm:block">{{ mutation.data.moment.toDateString() }}</span>
+          <span class="hidden sm:block">{{ mutation.data.moment.toLocaleDateString($i18n.locale, {
+            dateStyle: 'full'
+          }) }}</span>
           <span class="sm:hidden">{{
-            mutation.data.moment.toLocaleDateString('nl-NL', {
```
Comment (SuperVK) [inline src/components/Mutations/MutationsTableComponent.vue]: not sure if I fully agree with this, in English mode (for 90% of the users), the date will be American, which is date / month / year, which is super confusing. Maybe "correct" as you have selected that locale, but for most of our userbase not the preferred option.

File: src/main.ts [hunk 2]
```diff
@@ -41,11 +38,13 @@ import Skeleton from "primevue/skeleton";
 import IconField from "primevue/iconfield";
 import InputIcon from "primevue/inputicon";
 import ProgressSpinner from "primevue/progressspinner";
+import ToggleButton from "primevue/togglebutton";
+import { setLocale as yupSetLocale } from 'yup';
 const app = createApp(App);
 
 
 const i18n = createI18n({
-    locale: localStorage.getItem('locale') || 'en',
```
Comment (SuperVK) [inline src/main.ts:49]: Don't agree, main language of GEWIS is english, we don't want internationals searching on how to change the language

**Review by SuperVK — CHANGES_REQUESTED** [review]

**Comment by SuperVK** [comment]
does not fully fix #158 
image

File: src/components/Mutations/MutationsTableComponent.vue (hunk 1 above)
Comment (tomudding) [reply src/components/Mutations/MutationsTableComponent.vue]: I agree with Victor that the `short` variant should be `YYYY-MM-DD`.

Comment (CodeNamedRobin) [reply src/components/Mutations/MutationsTableComponent.vue]: Fixed!

File: src/main.ts (hunk 2 above)
Comment (CodeNamedRobin) [reply src/main.ts:49]: Fixed

**Comment by CodeNamedRobin** [comment]
> does not fully fix #158 image

This is now fixed!

**Comment by SuperVK** [comment]
image
This does not load right now :(

**Review by SuperVK — APPROVED** [review]

**Pull request merged**